source/main.cc

```diff
@@ -408,6 +408,20 @@ void print_version_information(Stream &stream)
 }
 
 
+// hook into SIGABRT and kill off the program
+void signal_handler(int signal)
+{
+  if (signal == SIGABRT)
+    {
+      std::cerr << "SIGABRT received\n";
+    }
+  else
+    {
+      std::cerr << "Unexpected signal " << signal << " received\n";
+    }
+  std::_Exit(EXIT_FAILURE);
```

Comment:
Yeah, sadly it is c++11 only. I need something more brutal than ``std::abort`` because this causes the deadlock. Should I just call it conditionally and not worry about pre cxx11?